draco/analysis/flagging.py

```diff
@@ -1768,6 +1785,34 @@ def _static_rfi_mask_hook(self, freq, timestamp=None):
         """
         return np.ones_like(freq, dtype=bool)
 
+    def _mask_1d(self, rad, mask):
+        """Mask based on the median over time."""
+        y = np.ascontiguousarray(rad.astype(np.float64))
+        w = np.ascontiguousarray((~mask).astype(np.float64))
+
+        medt_y = weighted_median.quantile(y, w, self.quantile_1d)
+        medt_w = np.any(w, axis=-1).astype(np.float64)
+
+        if self.win_f_1d is None:
+            medf_medt_y = weighted_median.weighted_median(medt_y, medt_w)
+        else:
+            medf_medt_y = weighted_median.moving_weighted_median(
+                medt_y, medt_w, self.win_f_1d
+            )
+
+        absd_medt_y = np.abs(medt_y - medf_medt_y)
+
+        if self.win_f_1d is None:
+            mad_1d = self.MAD_TO_RMS * weighted_median.weighted_median(
+                absd_medt_y, medt_w
+            )
+        else:
+            mad_1d = self.MAD_TO_RMS * weighted_median.moving_weighted_median(
+                absd_medt_y, medt_w, self.win_f_1d
+            )
+
+        return absd_medt_y > (self.nsigma_1d * mad_1d), medt_y
+
     def _apply_sir(self, mask, baseflag, eta=0.2):
```

Comment:
I'm curious about using the scale invariant rank operator over the time axis only. I think this would effectively extend frequency bands that are masked by the sum threshold and not by the MAD, while not increasing the amount of masked data too significantly. This is what's done in the second stage of the visibility-based masking, and using the operator in time only seemed to only increase the amount of masked data by around 1%.

Reply:
Thanks!  I gave this a try and it properly masked most of the residual RFI during transits that was not being picked up by the MAD algorithm.  However, it increased the total amount of masked data by 3%, most of which was obviously good data.

I ended up applying the SIR operator but then only using the extended mask during the bright source transits.  This mostly does what I want and increases the amount of data masked by less than 1%.